src/nvim/channel.c

```diff
@@ -634,10 +627,12 @@ static void on_channel_output(Stream *stream, Channel *chan, RBuffer *buf,
   }
 
   rbuffer_consumed(buf, count);
-  if (reader->buffered) {
-    ga_concat_len(&reader->buffer, ptr, count);
-  } else if (callback_reader_set(*reader)) {
-    process_channel_event(chan, &reader->cb, type, ptr, count, 0);
+  // if buffer wasn't consumed, a pending callback is stalled. Amalgate the
```

Comment:
I think you were going for "amalgamate" instead of "amalgate" here, but "aggregate" is probably better.